File: TeddyPet_BE/src/main/java/fpt/teddypet/application/dto/response/UserProfileResponse.java
package fpt.teddypet.application.dto.response;

import fpt.teddypet.domain.enums.GenderEnum;
import fpt.teddypet.domain.enums.UserStatusEnum;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.UUID;

/**
 * Response DTO containing user profile information.
 * Used for the /me endpoint to get current authenticated user info.
 */
public record UserProfileResponse(
        UUID id,
        String username,
        String email,
        String firstName,
        String lastName,
        String phoneNumber,
        String avatarUrl,
        String altImage,
        GenderEnum gender,
        LocalDate dateOfBirth,
        LocalDateTime createdAt,
        UserStatusEnum status,
        String role) {
}

File: teddypet_mobile/lib/data/models/response/user/user_profile_response.dart
class UserProfileResponse {
  final String id;
  final String username;
  final String email;
  final String firstName;
  final String lastName;
  final String? phoneNumber;
  final String? avatarUrl;
  final String? altImage;
  final String? gender;
  final String? dateOfBirth;
  final String? createdAt;
  final String? status;
  final String? role;

  UserProfileResponse({
    required this.id,
    required this.username,
    required this.email,
    required this.firstName,
    required this.lastName,
    this.phoneNumber,
    this.avatarUrl,
    this.altImage,
    this.gender,
    this.dateOfBirth,
    this.createdAt,
    this.status,
    this.role,
  });

  factory UserProfileResponse.fromJson(Map<String, dynamic> json) {
    return UserProfileResponse(
      id: json['id']?.toString() ?? '',
      username: json['username'] ?? '',
      email: json['email'] ?? '',
      firstName: json['firstName'] ?? '',
      lastName: json['lastName'] ?? '',
      phoneNumber: json['phoneNumber'],
      avatarUrl: json['avatarUrl'],
      altImage: json['altImage'],
      gender: json['gender'],
      dateOfBirth: json['dateOfBirth'],
      createdAt: json['createdAt'],
      status: json['status'],
      role: json['role'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'username': username,
      'email': email,
      'firstName': firstName,
      'lastName': lastName,
      'phoneNumber': phoneNumber,
      'avatarUrl': avatarUrl,
      'altImage': altImage,
      'gender': gender,
      'dateOfBirth': dateOfBirth,
      'createdAt': createdAt,
      'status': status,
      'role': role,
    };
  }
}
